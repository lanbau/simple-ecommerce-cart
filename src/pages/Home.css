.book {
    width: 200px;
    border: 1px solid rgba(196, 182, 182, 0.253);
    padding: 20px;
    margin: 10px;
}

.books {
    display: flex;
    flex-wrap: wrap;
    padding: 20px;
    justify-content: center;
}

.book-src {
    padding-bottom: 16px;
}

.book-title {
    font-size: 22px;
    font-weight: bold;
    padding-bottom: 16px;
    
}
.book-title > a {
    color: rgba(44, 44, 44, 0.849);
    text-decoration: none;
} 

.book-title > a:hover {
    color: black;
} 

.book-description {
    padding-bottom: 16px;
}

.book-buy-button {
    background-color: pink;
    width: 100px;
    text-align: center;
    padding: 10px;
    cursor: pointer;
    margin-right: 16px;
}

.book-buy-button:hover {
    opacity: 0.7;
}