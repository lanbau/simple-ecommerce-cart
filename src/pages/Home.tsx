import React from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { setBook } from '../store/actions'
import { getBooks } from '../selectors';
import { BookInterface } from '../interfaces'

import './Home.css';
import {
    BrowserRouter as Router,
    Switch,
    Route,
    Link
  } from "react-router-dom";

const Home = () => {
    const dispatch = useDispatch()
    const books = useSelector(getBooks)
    // const { books } = useSelector((state: RootState) => state)
    const handleBook = (book: BookInterface) => {
        dispatch(setBook(book))
    }
    return <div className="books">
        {books?.map( book => {
            const { id, title, body, src } = book
            return (
                <div className="book" key={id}>
                    <div className="book-src"><img src={src} /></div>
                    <div className="book-title" onClick={() => handleBook(book)}>
                        <Link to="/book">{title}</Link>
                    </div>
                    <div className="book-description">{body}</div>
                    <div className="book-buy-button">Buy Now</div>
                </div>
            )
        })}
    </div>
}

export default Home