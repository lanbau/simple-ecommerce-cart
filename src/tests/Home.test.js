import React from 'react'
import { useSelector } from 'react-redux'
import Home from '../pages/Home'
import { MemoryRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'

jest.mock('react-redux', () => ({
  useDispatch: jest.fn(),
  useSelector: jest.fn(),
}))

describe('Home Page With Books', () => {
  test('renders books', () => {
    useSelector.mockImplementation((selector) => selector({
      books: [{
          userId: 1,
          id: 1,
          title: 'sunt aut facere repellat provident occaecati excepturi optio reprehenderit',
          body: 'quia et suscipit\nsuscipit recusandae consequuntur expedita et cum\nreprehenderit molestiae ut ut quas totam\nnostrum rerum est autem sunt rem eveniet architecto',
          price: 15,
          src: 'https://picsum.photos/200/300?random=1'
      }]
    }))
    const container = renderer
      .create(
          <MemoryRouter>
            <Home />
          </MemoryRouter>
      )
      .toJSON()
    expect(container).toMatchSnapshot()
  })
})