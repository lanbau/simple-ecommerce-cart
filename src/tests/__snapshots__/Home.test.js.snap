// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page With Books renders books 1`] = `
<div
  className="books"
>
  <div
    className="book"
  >
    <div
      className="book-src"
    >
      <img
        src="https://picsum.photos/200/300?random=1"
      />
    </div>
    <div
      className="book-title"
      onClick={[Function]}
    >
      <a
        href="/book"
        onClick={[Function]}
      >
        sunt aut facere repellat provident occaecati excepturi optio reprehenderit
      </a>
    </div>
    <div
      className="book-description"
    >
      quia et suscipit
suscipit recusandae consequuntur expedita et cum
reprehenderit molestiae ut ut quas totam
nostrum rerum est autem sunt rem eveniet architecto
    </div>
    <div
      className="book-buy-button"
    >
      Buy Now
    </div>
  </div>
</div>
`;
